File: Lib/Telerik/Examples/QuickStart/SyntaxEditor/FirstLook/ProjectFiles/StyleSelectors/RowStyleSelector.vb
Imports System.Linq
Imports System.Windows
Imports Telerik.Windows.Controls
Imports Telerik.Windows.Controls.GridView
Imports System.Windows.Controls

Namespace Telerik.Windows.Examples.GridView.FirstLook
	Public Class RowStyleSelector
		Inherits StyleSelector

		Public Overrides Function SelectStyle(ByVal item As Object, ByVal container As DependencyObject) As Style
			If CType(container, GridViewRow).GridViewDataControl.Items.IndexOf(item) = 0 Then
				Dim style As New Style(GetType(GridViewRow)) With {.BasedOn = DirectCast(Application.Current.Resources("GridViewRowStyle"), Style)}
				Dim setter As New Setter(GridViewRow.DetailsVisibilityProperty, Visibility.Visible)
				style.Setters.Add(setter)
				Return style
			End If

			Return New Style(GetType(GridViewRow)) With {.BasedOn = DirectCast(Application.Current.Resources("GridViewRowStyle"), Style)}
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/SyntaxEditor/FirstLook/ProjectFiles/StyleSelectors/RowStyleSelector.cs
using System.Linq;
using System.Windows;
using Telerik.Windows.Controls;
using Telerik.Windows.Controls.GridView;
using System.Windows.Controls;

namespace Telerik.Windows.Examples.GridView.FirstLook
{
    public class RowStyleSelector : StyleSelector
    {
        public override Style SelectStyle(object item, DependencyObject container)
        {
            if (((GridViewRow)container).GridViewDataControl.Items.IndexOf(item) == 0)
            {
				Style style = new Style(typeof(GridViewRow)) { BasedOn = (Style)Application.Current.Resources["GridViewRowStyle"] };
                Setter setter = new Setter(GridViewRow.DetailsVisibilityProperty, Visibility.Visible);
                style.Setters.Add(setter);
                return style;
            }

            return new Style(typeof(GridViewRow)) { BasedOn = (Style)Application.Current.Resources["GridViewRowStyle"] };
        }
    }
}
